pmd-core/src/main/java/net/sourceforge/pmd/lang/ast/AbstractNode.java

```diff
@@ -425,22 +425,91 @@ public void remove() {
         // Detach current node of its parent, if any
         final Node parent = jjtGetParent();
         if (parent != null) {
-            parent.removeChildAtIndex(jjtGetChildIndex());
-            jjtSetParent(null);
+            parent.remove(jjtGetChildIndex());
         }
+    }
+
+    @Override
+    public void remove(final int index) {
+        if (0 > index || index >= jjtGetNumChildren()) {
+            return;
+        }
+
+        // Null child may have been caused due to an invalid insertion/addition
+        final Node oldChild = Objects.requireNonNull(children[index]);
 
-        // TODO [autofix]: Notify action for handling text edition
+        // Remove the child at the given index
+        children = ArrayUtils.remove(children, index);
+        // Update the remaining & left-shifted children indexes
+        for (int i = childIndex; i < jjtGetNumChildren(); i++) {
+            jjtGetChild(i).jjtSetChildIndex(i);
+        }
+
+        // Detach old child node of its parent, if any
+        oldChild.jjtSetParent(null);
+
+        // Finally, report the remove event
+        // removeChildEvent(this, oldChild, index); // TODO [autofix]
     }
 
     @Override
-    public void removeChildAtIndex(final int childIndex) {
```

Comment:
removing this method is a breaking API change. We should readd it, have it `@Deprecated` and internally call to  the new `remove` method if we want to rename it

Reply:
Thanks for the advice. I'll apply this change too.